src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -103,13 +104,16 @@ def convert_sf_to_sp_type(
         raise ValueError("Negative value is not a valid input for StringType")
     if column_type_name == "TIME":
         return TimeType()
-    if column_type_name in (
-        "TIMESTAMP",
-        "TIMESTAMP_LTZ",
-        "TIMESTAMP_TZ",
-        "TIMESTAMP_NTZ",
-    ):
-        return TimestampType()
+    if column_type_name.startswith("TIMESTAMP"):
```

Comment:
Is this `if` needed? Should we directly check column type instead? The code duplication seems to be minimal.

Reply:
Sure.